kyaml/kio/byteio_reader.go

```diff
@@ -115,11 +123,32 @@ func (r *ByteReader) Read() ([]*yaml.RNode, error) {
 	if err != nil {
 		return nil, errors.Wrap(err)
 	}
-	values := strings.Split(input.String(), "\n---\n")
+	inputStr := input.String()
+
+	// check if json is accepted and if input bytes are in json format
+	if r.AcceptJSON && yaml.ValidJSON(input.String()) {
+		// convert json to yaml string to generate resource list object
+		// with appropriate format annotation
+		yamlValue, err := yaml.ConvertJSONToYAMLString(input.String())
```

Comment:
Do we need this?  What happens if just let the YAML parser parse the JSON?

Reply:
I tested it and it is not parsing json as expected.

Reply:
Weird.  What is happening?